ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -244,6 +244,23 @@ object Queries {
     }
   }
 
+  private[http] def fetchById(
+      parties: OneAnd[Set, String],
+      tpid: SurrogateTpId,
+      contractId: String)(
+      implicit log: LogHandler,
+      gvs: Get[Vector[String]],
+      pvs: Put[Vector[String]])
+    : ConnectionIO[Option[DBContract[Unit, JsValue, JsValue, Vector[String]]]] =
+    selectContracts(parties, tpid, sql"contract_id = $contractId").option
```

Comment:
Should we create indices on `contract_id` and maybe `key` by default? Seems common enough that we might not want to leave this to users.

Reply:
`contract_id` is already a primary key. For contract keys, 18e80597c7a74d0a311b621ae3d63fbe6684b2a7